datasetinsights/io/source/base.py

```diff
@@ -0,0 +1,42 @@
+import re
+from abc import ABCMeta, abstractmethod
+
+
+class DownloaderRegistry(ABCMeta):
```

Comment:
The downloaders are a typical Factory/Strategy pattern. Can you rename them accordingly?

Reply:
Do you want to rename the DownloaderRegistry or the new_cls variable?

Reply:
`DownloaderRegistry` implies it's a registry which allow users to find a dataset location given a name (or unique string). We already abandoned `registry` approach during the design review, and I think it should be renamed accordingly to reduce confusion.